common/lib/common-utils/src/bufferBrowser.ts

```diff
@@ -92,10 +92,8 @@ export class IsoBuffer extends Uint8Array {
             return IsoBuffer.fromString(value, encodingOrOffset as string | undefined);
         // Capture any typed arrays, including Uint8Array (and thus - IsoBuffer!)
         } else if (value !== null && typeof value === "object" && isArrayBuffer(value.buffer)) {
-            // Support currently for full array, no view ports! (though it can be added in future)
-            assert(value.byteOffset === 0, 0x000 /* "nonzero isobuffer byte offset" */);
-            assert(value.byteLength === value.buffer.byteLength, 0x001 /* "unexpected isobuffer byte length" */);
-            return IsoBuffer.fromArrayBuffer(value.buffer, encodingOrOffset as number | undefined, length);
+            return IsoBuffer.fromArrayBuffer(
+                Uint8ArrayToArrayBuffer(value), encodingOrOffset as number | undefined, length);
```

Comment:
this should be done using arithmetic with the offset and length instead of calling the conversion which might do an array copy

Reply:
So when I convert using "Uint8ArrayToArrayBuffer" it used the offset and length to calculate the arrayBuffer. The encodingOrOffset and length in this line will then maybe specify the offset and length within that calculated array buffer.

Reply:
I believe the code is correct, but not efficient (in general case). I.e. what @heliocliu is trying to say (if I got it right) is - we may combine office from within original typed array and encodingOrOffset and create Uint8array without every slicing (i.e. it's just a viewport into original array, as long as we correctly calculated offset & length).

Note though that it's complicated by encodingOrOffset potentially having an encoding (not offset).
Though fromArrayBuffer() disregards it completely, so I think an assert would be great to validate its type is a number (if we do not support encodings, I'm not even sure what it means to use different encodings here).

Reply:
Right, slice will create a new array and copy the contents which we don't want to do.  `encodingOrOffset` is just an ugly version of method overloading because I couldn't get typescript to be happy with overloaded declarations 🙄 so this is not expected to ever be an encoding in the Node Buffer API we're mimicking

Reply:
I reverted my changes here for this PR, since I want to get this in. The node behavior and browser didn't match so have to dig deeper what is the issue.

Reply:
In node, res = BufferNode.IsoBuffer.from(x, 0, 4) where x.byteOffset = 0 and x.bytelength = 10, returns same buffer where res.byteoffset = 0 and res.bytelength = 10